application/services/Shipments.php

```diff
@@ -944,6 +944,83 @@ public function addShipment($params) {
             $distroService->updateDistributionStatus($params['distribution'], 'pending');
         }
     }
+    public function addShipmentagainstditribution($params,$distribution) {
+        $authNameSpace = new Zend_Session_Namespace('administrators');
+        $db = new Application_Model_DbTable_Shipments();
+
+        $distroService = new Application_Service_Distribution();
+        $distro = $distroService->getDistribution($distribution);
+
+        $controlCount = 0;
+        if (isset($params['control'])) {
+            foreach ($params['control'] as $control) {
+                if ($control == 1) {
+                    $controlCount+=1;
+                }
+            }
+        }
+        $data = array(
+            'shipment_code' => $params['shipmentCode'],
+            'distribution_id' => $distro['distribution_id'],
+            'scheme_type' => 'tb',
+            'shipment_date' => $distro['distribution_date'],
+            'number_of_samples' => count($params['sampleName']) - $controlCount,
+            'number_of_controls' => $controlCount,
+            'lastdate_response' => Application_Service_Common::ParseDate($params['lastDate']),
+            'is_official' => $params['isOfficial'] == 'yes' ? 1 : 0,
+            'created_on_admin' => new Zend_Db_Expr('now()'),
+            'created_by_admin' => $authNameSpace->primary_email
+        );
+        if (isset($params['isFollowUp'])) {
+            $data['follows_up_from'] = $params['followsUpFrom'];
+        }
+        $dbAdapter = Zend_Db_Table_Abstract::getDefaultAdapter();
+        $lastId = $db->insert($data);
+        if (isset($params['autoEnroll'])) {
+            $participantService = new Application_Service_Participants();
+            $previouslyEnrolledParticipants = $participantService->getEnrolledByShipmentId($params['followsUpFrom']);
+            $enrollmentTable = new Application_Model_DbTable_ShipmentParticipantMap();
+            $enrollmentTable->shipItNow(array(
+                'shipmentId' => $lastId,
+                'participants' => array_map(create_function('$p', 'return $p["participant_id"];'), $previouslyEnrolledParticipants)
+            ));
+        }
+        $size = count($params['sampleName']);
+        if ($params['schemeId'] == 'tb') {
```

Comment:
We have removed all other schemes so checking the schemeId is not necessary. You can remove the if statement here.